src/components/ChannelPreview/ChannelPreview.js

```diff
@@ -4,24 +4,29 @@ import PropTypes from 'prop-types';
 import { ChatContext } from '../../context';
 import { useLatestMessagePreview } from './hooks/useLatestMessagePreview';
 
-const ChannelPreviewWithContext = React.memo((props) => {
-  const { channel, client } = props;
+const ChannelPreview = (props) => {
+  const { channel } = props;
+  const { client } = useContext(ChatContext);
   const [lastMessage, setLastMessage] = useState({});
   const [unread, setUnread] = useState(channel.countUnread());
   const latestMessage = useLatestMessagePreview(channel, lastMessage);
 
   useEffect(() => {
-    const handleNewMessageEvent = (e) => {
+    const handleEvent = (e) => {
       setLastMessage(e.message);
       setUnread(channel.countUnread());
     };
 
-    channel.on('message.new', handleNewMessageEvent);
+    channel.on('message.new', handleEvent);
+    channel.on('message.updated', handleEvent);
+    channel.on('message.deleted', handleEvent);
 
     return () => {
-      channel.off('message.new', handleNewMessageEvent);
+      channel.off('message.new', handleEvent);
+      channel.off('message.updated', handleEvent);
+      channel.off('message.deleted', handleEvent);
     };
-  });
+  }, []);
```

Comment:
@vishalnarkhede you and @DanC5 figure this out yesterday on the necessity of this empty dep array? thought someone mentioned events not being triggered

Reply:
Nahh that one resolved!!